codec-dns/src/main/java/io/netty/handler/codec/dns/TcpDnsQueryDecoder.java

```diff
@@ -0,0 +1,81 @@
+package io.netty.handler.codec.dns;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.handler.codec.CorruptedFrameException;
+import io.netty.handler.codec.LengthFieldBasedFrameDecoder;
+
+public final class TcpDnsQueryDecoder extends LengthFieldBasedFrameDecoder {
+    private final DnsRecordDecoder decoder;
+
+    public TcpDnsQueryDecoder() {
+        this(DnsRecordDecoder.DEFAULT, 64 * 1024);
+    }
+
+    public TcpDnsQueryDecoder(DnsRecordDecoder decoder, int maxFrameLength) {
+        super(maxFrameLength, 0, 2, 0, 2);
+        this.decoder = decoder;
+    }
+
+    @Override
+    protected Object decode(ChannelHandlerContext ctx, ByteBuf in) throws Exception {
+        ByteBuf frame = (ByteBuf) super.decode(ctx, in);
+        if (frame == null) {
+            return null;
+        }
+
+        ByteBuf buf = frame.slice();
+        DnsQuery query = newQuery(buf);
+        boolean success = false;
+        try {
+            int questionCount = buf.readUnsignedShort();
+            int answerCount = buf.readUnsignedShort();
+            int authorityRecordCount = buf.readUnsignedShort();
+            int additionalRecordCount = buf.readUnsignedShort();
+            this.decodeQuestions(query, buf, questionCount);
+            this.decodeRecords(query, DnsSection.ANSWER, buf, answerCount);
+            this.decodeRecords(query, DnsSection.AUTHORITY, buf, authorityRecordCount);
+            this.decodeRecords(query, DnsSection.ADDITIONAL, buf, additionalRecordCount);
```

Comment:
nit: you can remove all the `this.` above

Reply:
removed all the `this.`